geomstats/geometry/pullback_metric.py

```diff
@@ -0,0 +1,94 @@
+"""Classes for the pullback metric."""
+
+import autograd
+
+import geomstats.backend as gs
+from geomstats.geometry.euclidean import EuclideanMetric
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class PullbackMetric(RiemannianMetric):
+    r"""Pullback metric.
+
+    Let :math:`f` be an immersion :math:`f: M \rightarrow N`
+    of one manifold :math:`M` into the Riemannian manifold :math:`N`
+    with metric :math:`g`.
+    The pull-back metric :math:`f^*g` is defined on :math:`M` for a
+    base point :math:`p` as:
+    :math:`(f^*g)_p(u, v) = g_{f(p)}(df_p u , df_p v)
+    \quad \forall u, v \in T_pM`
+
+    Note
+    ----
+    The pull-back metric is currently only implemented for an
+    immersion into the Euclidean space, i.e. for
+    :math:`N=\mathbb{R}^n`.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension of the underlying manifold.
+    embedding_dim : int
+        Dimension of the embedding Euclidean space.
+    immersion : callable
+        Map defining the immersion into the Euclidean space.
+    """
+
+    def __init__(
+            self, dim, embedding_dim, immersion,
+            jacobian_immersion=None, tangent_immersion=None):
+        super(PullbackMetric, self).__init__(dim=dim)
+        self.embedding_metric = EuclideanMetric(embedding_dim)
+        self.immersion = immersion
+        if jacobian_immersion is None:
+            jacobian_immersion = autograd.jacobian(immersion)
```

Comment:
This can work with other backends with `gs.autograd.jacobian`

Reply:
Good point, thanks!